mixer/pkg/protobuf/yaml/dynamic/auth_test.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamic
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/gogo/protobuf/types"
+
+	"istio.io/api/mixer/adapter/model/v1beta1"
+	attributeV1beta1 "istio.io/api/policy/v1beta1"
+	policypb "istio.io/api/policy/v1beta1"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/metric"
+	spy "istio.io/istio/mixer/test/spybackend"
+)
+
+type AuthMode int
+
+const (
+	PLAINTEXT AuthMode = 0
+	TLS       AuthMode = 1
+	MTLS      AuthMode = 2
+)
+
+func TestAuth(t *testing.T) {
+	// prep for test.
+	// write token file.
+	tokenPath := path.Join(os.TempDir(), "token")
+	writeTestToken(t, tokenPath)
+	defer func() {
+		_ = os.Remove(tokenPath)
+	}()
+	// start oauth server
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("access_token=test-token&token_type=bearer"))
```

Comment:
nit: would be nice to verify OAuth request

Reply:
Done.